File: leetcode/TwoSum/TwoSum.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
  indices = []
  map = {}
  i = 0
  nums.each do |e|
      map[i] = e
      i+=1
  end

  map.each do |k, v|
  diff = target - v
      if map.value?(diff) && map.key(diff) != k
          indices = [k, map.key(diff)]
          break
      end
  end
  indices
end

File: leetcode/TwoSum/TwoSum.go
func twoSum(nums []int, target int) []int {
	numsComplementOfTargetMap := make(map[int]int)
	for index, element := range nums {
		numsComplementOfTargetMap[target-element] = index
	}

	for index, element := range nums {
		value, ok := numsComplementOfTargetMap[element]
		if ok && index != value {
			return []int{numsComplementOfTargetMap[element], index}
		}
	}
	return []int{}
}